src/highlight/highlight.rs

```diff
@@ -1024,7 +1024,7 @@ impl<'s> Highlighter<'s> {
         let mut is_set = expanded_cmd == "set";
         // If we have seen a "--" argument, color all options from then on as normal arguments.
         let mut have_dashdash = false;
-        for v in &stmt.args_or_redirs {
+        for v in &stmt.args_or_redirs[..] {
```

Comment:
can you extract some more of the unrelated changes to separate PRs?
BTW we also bumped the MSRV, so the changes in a0d9c23d6eee03d0851078103aa0bb7bf2e83d98 (ast: fix MSRV breakage, 2025-09-11) are probably no longer needed.